tests/saispy.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "saitypes.h"
+
+// Spy C functin pointer to std::function to access closure
+// Internal using static `spy` function pointer to invoke std::function `fake`
```

Comment:
I read the comment block and also the use case in aclorch_ut.cpp, still not get the design idea of SpyOn. Could you elaborate? We expected a design doc and/or community talk.

Reply:
I've put the unit test for how to use the code in `saispy_ut.cpp` can you check it first ?